File: Back/src/SistemaEmpresas.Domain/Entities/Empresa.cs
namespace SistemaEmpresas.Domain.Entities;

public class Empresa
{
    public Guid Id { get; set; }
    public required string RazaoSocial { get; set; }    
    public required string Cnpj { get; set; }
    public bool Status { get; set; }
    public required Endereco Endereco { get; set; }
    public required Contato Contato { get; set; }
}

File: Front/SistemaEmpresas-App/src/app/features/empresa/models/empresa.ts
import { Contato } from "./contato";
import { Endereco } from "./endereco";

export interface Empresa {
    id: string;
    razaoSocial: string;
    cnpj: string;
    ativo: boolean;
    
    endereco: Endereco;
    contato: Contato;
}